virtual_rainforest/models/animals/__init__.py

```diff
@@ -27,7 +27,4 @@
   the animal module.
 """  # noqa: D205, D415
 
-from virtual_rainforest.core.base_model import register_model
-from virtual_rainforest.models.animals.animal_model import AnimalModel
-
-register_model(__name__, AnimalModel)
+from virtual_rainforest.models.animals.animal_model import AnimalModel  # noqa: F401
```

Comment:
Same question for `AnimalModel` does it still need to be imported (and I assume for every other model `__init__.py`)?

Reply:
That is basically my point of concern above - I've moved _all_ of the registration process inside the config `build_schema` method, and maybe that should just import the required modules and leave the `register_module` call in the model `__init__` so that they also get registered when imported?

Reply:
I think it might be nice to make this more explicit and keep the call to `register_model` (or whatever you want to call it).

I'd also make the model class type an argument, so that users have to actually *use* the imported class somewhere. This means a) they don't have to suppress any warnings (which is a bit gross) and b) they won't forget to import the model class, because they need it as an argument and c) it's obvious that the purpose of importing the class is for model registration.

Reply:
I'm not so bothered about all of that - the documentation is clear on what is needed and why - but I think the main thing for restoring that structure  is that it us just much cleaner if importing a component of a model triggers registration, as opposed to then having to run `register_module` as a second step.